### PR TITLE
Add a test that always publishes in Debug

### LINKED ISSUE #99: Add tests to verify tests work for both Debug and Release configurations
Right now we publish test apps in the configuration tests are running. This doesn't give us enough coverage for `Debug` publishes in particularly on the CI where it always builds in `Release`. Based on https://github.com/aspnet/MvcPrecompilation/pull/97#discussion_r105547719, let's add some tests that verify debug and release configurations all the time.

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #99

### CODE REVIEW

**Comment by pranavkm** [comment]
Bump

File: test/FunctionalTests/SimpleAppTest_CoreCLR.cs [hunk 1]
```diff
@@ -38,5 +39,32 @@ public class SimpleAppTest_CoreCLR :
...
+        public async Task Precompilation_PublishesPdbsToOutputDirectory()
+        {
+            using (StartLog(out var loggerFactory))
+            {
+                // Arrange
+                var deployment = await Fixture.CreateDeploymentAsync(loggerFactory);
+                var pdbPath = Path.Combine(deployment.ContentRoot, Fixture.ApplicationName + ".PrecompiledViews.pdb");
+
+                // Act & Assert
+                Assert.True(File.Exists(pdbPath), $"PDB at {pdbPath} was not found.");
```
Comment (ajaybhargavb) [inline test/FunctionalTests/SimpleAppTest_CoreCLR.cs]: Shouldn't this say "PDB at {pdbPath} was ~~not~~ found."?

File: test/FunctionalTests/SimpleAppTest_Desktop.cs [hunk 2]
```diff
@@ -41,5 +42,32 @@ public class SimpleAppTest_Desktop :
...
+        public async Task Precompilation_PublishesPdbsToOutputDirectory()
+        {
+            using (StartLog(out var loggerFactory))
+            {
+                // Arrange
+                var deployment = await Fixture.CreateDeploymentAsync(loggerFactory);
+                var pdbPath = Path.Combine(deployment.ContentRoot, Fixture.ApplicationName + ".PrecompiledViews.pdb");
+
+                // Act & Assert
+                Assert.True(File.Exists(pdbPath), $"PDB at {pdbPath} was not found.");
```
Comment (ajaybhargavb) [inline test/FunctionalTests/SimpleAppTest_Desktop.cs]: Ditto

File: test/FunctionalTests/SimpleAppTest_CoreCLR.cs (hunk 1 above)
Comment (pranavkm) [inline test/FunctionalTests/SimpleAppTest_CoreCLR.cs]: The string is printed when the assertion fails. In both of these cases, the pdb could not be found.

Comment (ajaybhargavb) [inline test/FunctionalTests/SimpleAppTest_CoreCLR.cs]: Ah got it. 👍

**Pull request merged**